skrifa/src/scale/cff/hint.rs

```diff
@@ -904,6 +907,182 @@ fn msb_mask(bit: usize) -> u8 {
     1 << (7 - (bit & 0x7))
 }
 
+pub(super) struct HintingSink<'a, S> {
+    state: &'a HintState,
+    sink: &'a mut S,
+    stem_hints: [StemHint; MAX_HINTS],
+    stem_count: u8,
+    mask: HintMask,
+    initial_map: HintMap,
+    map: HintMap,
+    /// Most recent move_to in character space.
+    start_point: Option<[Fixed; 2]>,
+    /// Most recent line_to. First two elements are coords in character
+    /// space and the last two are in device space.
+    pending_line: Option<[Fixed; 4]>,
+}
+
+impl<'a, S: CommandSink> HintingSink<'a, S> {
+    pub fn new(state: &'a HintState, sink: &'a mut S) -> Self {
+        let scale = state.scale;
+        Self {
+            state,
+            sink,
+            stem_hints: [StemHint::default(); MAX_HINTS],
+            stem_count: 0,
+            mask: HintMask::all(),
+            initial_map: HintMap::new(scale),
+            map: HintMap::new(scale),
+            start_point: None,
+            pending_line: None,
+        }
+    }
+
+    pub fn finish(&mut self) {
+        self.maybe_close_subpath();
+    }
+
+    fn maybe_close_subpath(&mut self) {
+        // This requires some explanation. The hint mask can be modified
+        // during charstring evaluation which changes the set of hints that
+        // are applied. FreeType ensures that the closing line (this logic
+        // doesn't apply to cubics *shrug*) for any subpath is transformed
+        // with the same hint map as the starting point for the subpath.
```

Comment:
"for any subpath is transformed" is missing some word somewhere

Reply:
The parenthetical seems to completely destroy the flow of this sentence. I’ll clean it up.